CMakeLists.txt

```diff
@@ -57,48 +57,66 @@ set(PROJECT_VERSION_PATCH 0)
 
 if(IS_RELEASE_VERSION)
-  set(PROJECT_VERSION_TWEAK 0)
-  set(VERSION_REVISION "")
-elseif(EXISTS "${CMAKE_SOURCE_DIR}/.git")
-  find_package(Git)
-  if(Git_FOUND)
-    execute_process(
-      COMMAND git rev-list --count HEAD ^tags/v${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}
-      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-      OUTPUT_VARIABLE PROJECT_VERSION_TWEAK
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-    execute_process(
-      COMMAND git rev-parse --short --verify HEAD
-      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-      OUTPUT_VARIABLE VERSION_COMMIT_HASH
-      OUTPUT_STRIP_TRAILING_WHITESPACE
-    )
-    execute_process(
-      COMMAND git diff-index --quiet HEAD --
-      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-      RESULT_VARIABLE VERSION_COMMIT_DIRTY
-    )
-    if(${VERSION_COMMIT_DIRTY})
-      set(VERSION_REVISION ".${PROJECT_VERSION_TWEAK}-${VERSION_COMMIT_HASH}-dirty")
+    set(PROJECT_VERSION_TWEAK 0)
+    set(VERSION_REVISION "")
+elseif (EXISTS "${CMAKE_SOURCE_DIR}/.git")
+    find_package(Git)
+    if(Git_FOUND)
+        execute_process(
+            COMMAND git rev-list --count HEAD ^tags/v${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+            OUTPUT_VARIABLE PROJECT_VERSION_TWEAK
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+        execute_process(
+            COMMAND git rev-parse --short --verify HEAD
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+            OUTPUT_VARIABLE VERSION_COMMIT_HASH
+            OUTPUT_STRIP_TRAILING_WHITESPACE
+        )
+        execute_process(
+            COMMAND git diff-index --quiet HEAD --
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+            RESULT_VARIABLE VERSION_COMMIT_DIRTY
+        )
+        if(${VERSION_COMMIT_DIRTY})
+            set(VERSION_REVISION ".${PROJECT_VERSION_TWEAK}-${VERSION_COMMIT_HASH}-dirty")
+        else()
+            set(VERSION_REVISION ".${PROJECT_VERSION_TWEAK}-${VERSION_COMMIT_HASH}")
+        endif()
     else()
-      set(VERSION_REVISION ".${PROJECT_VERSION_TWEAK}-${VERSION_COMMIT_HASH}")
+        set(PROJECT_VERSION_TWEAK 0)
+        set(VERSION_REVISION " unknown development version")
     endif()
-  else()
+else()
     set(PROJECT_VERSION_TWEAK 0)
     set(VERSION_REVISION " unknown development version")
-  endif()
-else()
-  set(PROJECT_VERSION_TWEAK 0)
-  set(VERSION_REVISION " unknown development version")
 endif()
 
-if (VITA_BUILD)
-    set(VITA_VERSION "${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}.${PROJECT_VERSION_PATCH}${VERSION_REVISION}")
+if(VITA_BUILD)
+    string(LENGTH "${PROJECT_VERSION_MAJOR}" NUM_DIGITS)
+    if(NUM_DIGITS LESS 1)
+        set(VITA_VERSION_MAJOR "00")
+    elseif(NUM_DIGITS EQUAL 1)
+        set(VITA_VERSION_MAJOR "0${PROJECT_VERSION_MAJOR}")
+    else()
+        string(SUBSTRING "${PROJECT_VERSION_MAJOR}" 0 2 VITA_VERSION_MAJOR)
```

Comment:
I suggest to error instead so there isn't weird unnoticed behavior down the road.

Reply:
fixed in 679aa8074db0489c1eb45a2b0befc36c9642fc00